src/main/java/co/cask/hydrator/plugin/spark/dynamic/ScalaSparkCodeExecutor.java

```diff
@@ -0,0 +1,395 @@
+      // Should already been checked in initialize. This is to safeguard in case the call sequence changed in future.
+      throw new IllegalArgumentException("Input schema must be provided for using DataFrame in Spark Compute");
+    }
+
+    SQLContext sqlContext = getSQLContext(context.getSparkContext().sc());
+
+    StructType rowType = DataFrames.toDataType(inputSchema);
+    JavaRDD<Row> rowRDD = javaRDD.map(new RecordToRow(rowType));
+
+    Object dataFrame = createDataFrame(sqlContext, rowRDD, rowType);
+    return takeContext ? method.invoke(null, dataFrame, context) : method.invoke(null, dataFrame);
+  }
+
+  private String generateSourceClass(String className) {
+    StringWriter writer = new StringWriter();
+
+    try (PrintWriter sourceWriter = new PrintWriter(writer, false)) {
+      sourceWriter.println("package " + className.substring(0, className.lastIndexOf('.')));
+      // Includes some commonly used imports.
+      sourceWriter.println("import co.cask.cdap.api.data.format._");
+      sourceWriter.println("import co.cask.cdap.api.data.schema._");
+      sourceWriter.println("import co.cask.cdap.etl.api.batch._");
+      sourceWriter.println("import org.apache.spark._");
+      sourceWriter.println("import org.apache.spark.api.java._");
+      sourceWriter.println("import org.apache.spark.rdd._");
+      sourceWriter.println("import org.apache.spark.sql._");
+      sourceWriter.println("import org.apache.spark.SparkContext._");
+      sourceWriter.println("import scala.collection.JavaConversions._");
+      sourceWriter.println("object " + className.substring(className.lastIndexOf('.') + 1) + " {");
+      sourceWriter.println(code);
+      sourceWriter.println("}");
+    }
+
+    return writer.toString();
+  }
+
+  /**
+   * Validates the given type is {@code RDD<StructuredRecord>}.
+   */
+  private void validateRDDType(Type rddType, String errorMessage) {
+    if (!(rddType instanceof ParameterizedType)) {
+      throw new IllegalArgumentException(errorMessage);
+    }
+    if (!RDD.class.equals(((ParameterizedType) rddType).getRawType())) {
+      throw new IllegalArgumentException(errorMessage);
+    }
+
+    Type[] typeParams = ((ParameterizedType) rddType).getActualTypeArguments();
+    if (typeParams.length < 1 || !typeParams[0].equals(StructuredRecord.class)) {
+      throw new IllegalArgumentException(errorMessage);
+    }
+  }
+
+  private SQLContext getSQLContext(SparkContext sc) {
+    SQLContext sqlContext = sqlContextThreadLocal.get();
+    if (sqlContext != null && !sqlContext.sparkContext().isStopped()) {
+      return sqlContext;
+    }
+
+    synchronized (this) {
+      sqlContext = sqlContextThreadLocal.get();
+      if (sqlContext == null || sqlContext.sparkContext().isStopped()) {
+        sqlContext = new SQLContext(sc);
+        sqlContextThreadLocal.set(sqlContext);
+      }
+    }
+
+    return sqlContext;
+  }
+
+  private Method getMethod(ClassLoader classLoader, String className) {
```

Comment:
this is the only part of this class that is not a straight copy-paste from the ScalaSparkCompute. It has been modified to support a different method name and different return type.